packages/editor/src/store/actions.js

```diff
@@ -236,14 +236,29 @@ export const savePost =
 				);
 		}
 
+		// Run the hook with legacy unstable name for backward compatibility
 		if ( ! error ) {
-			await applyFilters(
-				'editor.__unstableSavePost',
-				Promise.resolve(),
-				options
-			).catch( ( err ) => {
+			try {
+				await applyFilters(
+					'editor.__unstableSavePost',
+					Promise.resolve(),
+					options
+				);
```

Comment:
Doesn't seem like it'd be as effective as a deprecation message on a call to `addFilter`, but core has this PHP function, so replicating it might be the way to go - https://developer.wordpress.org/reference/functions/apply_filters_deprecated/

There's no JS equivalent unfortunately, so it'd need to be implemented in the hooks package, which makes this a bit more involved. 😅 

I guess the reason it's not handled on `add_filter` in core is the deprecation messages would have to be added within the `add_filter` function?

Reply:
Yes, `apply_filters_deprecated` is exactly what I need 🙂 It does nothing when no filter is registered, but complains otherwise.

It needs to be added to `@wordpress/hooks` first, but why not, I'm already adding [async filters](https://github.com/WordPress/gutenberg/pull/64204) anyway. Seems that the `hooks` package will get a nice upgrade in 6.7.

Reply:
I think a `apply_filters_deprecated` port to JS will make a lot of sense in the package, especially as we deprecate filters/actions in the future. Work for another PR of course 😉 It doesn't seem like the unstable hook was used anywhere (I've checked extensively on `wpdirectory.net` and a few more places)